src/rgw/rgw_http_client_curl.cc

```diff
@@ -0,0 +1,103 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+extern "C" {
+#include <curl/curl.h>
+}
+
+#include "rgw_http_client_curl.h"
+#include <mutex>
+#include <vector>
+
+#include "rgw_common.h"
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+#ifdef WITH_RADOSGW_OPENSSL_LOCK_COMPAT
+#include <openssl/crypto.h>
+#endif
+
+namespace rgw {
+namespace curl {
+
+#ifdef HAVE_CURL_MULTI_WAIT
```

Comment:
This is fine, but you might also consider moving the ifdef inside of the function; the functionality is the same but it's a bit more concise.